File: app/src/main/java/com/schrondinger/quin/base/mvp/BasePresenter.java
package com.schrondinger.quin.base.mvp;

import android.content.Context;

import com.schrongder.nba.common.RxManager;

/**
 * Created by hp on 2018/1/3.
 */

public abstract class BasePresenter<M,V> {
    public Context context;
    public M mModel;
    public V mView;
    public RxManager mRxManager = new RxManager();

    public void setVM(V v, M m) {
        this.mView = v;
        this.mModel = m;
    }


    public void onDestroy() {
        mRxManager.clear();
    }
}

File: app/src/main/java/com/schrondinger/quin/base/mvp/BasePresenter.kt
package com.schrondinger.quin.base.mvp

import android.arch.lifecycle.LifecycleOwner
import android.content.Context
import com.schrondinger.quin.common.RxManager


/**
 * Created by hp on 2018/1/3.
 */

abstract class BasePresenter<M, V> {
    var context: Context? = null
    var mModel: M? = null
    var mView: V? = null
    var mRxManager = RxManager()

    lateinit var lifecycleOwner: LifecycleOwner

    fun setVM(v: V?, m: M?) {
        this.mView = v
        this.mModel = m
    }


    fun onDestroy() {
        mRxManager.clear()
    }
}
